File: rust/tests/evaluator_runs_test.rs
use sikaru::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_evaluator_runs_create_evaluator_run_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .evaluator_runs
        .create_evaluator_run(
            &"project_id".to_string(),
            &CreateEvaluatorRunRequest {
                trace_ids: vec!["traceIds".to_string()],
                agent_id: None,
                backfill_window: None,
                dataset_name: None,
                evaluator_name: None,
                filter_expression: None,
                issue_id: None,
                issue_title: None,
                sample_rate: None,
                signal_tags: None,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count(
        "POST",
        "/v1/projects/project_id/evaluator-runs",
        None,
        1,
    )
    .await
    .unwrap();
}

File: ruby/test/wire/evaluator_runs_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class EvaluatorRunsWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      token: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_evaluator_runs_create_evaluator_run_with_wiremock
    test_id = "evaluator_runs.create_evaluator_run.0"

    @client.evaluator_runs.create_evaluator_run(
      project_id: "project_id",
      trace_ids: ["traceIds"],
      request_options: {
        additional_headers: {
          "X-Test-Id" => "evaluator_runs.create_evaluator_run.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/evaluator-runs",
      query_params: nil,
      expected: 1
    )
  end
end
